Mainnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -0,0 +1,639 @@
+        private set => this.PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    /// <summary>
+    /// Name for non-fungible token contract
+    /// </summary>
+    public string Name
+    {
+        get => this.PersistentState.GetString(nameof(Name));
+        private set => this.PersistentState.SetString(nameof(Name), value);
+    }
+
+    /// <summary>
+    /// Symbol for non-fungible token contract
+    /// </summary>
+    public string Symbol
+    {
+        get => this.PersistentState.GetString(nameof(Symbol));
+        private set => this.PersistentState.SetString(nameof(Symbol), value);
+    }
+
+    /// <summary>
+    /// The next token index which is going to be minted
+    /// </summary>
+    private ulong NextTokenId
+    {
+        get => this.PersistentState.GetUInt64(nameof(NextTokenId));
+        set => this.PersistentState.SetUInt64(nameof(NextTokenId), value);
+    }
+
+    private string GetTokenByIndexKey(ulong index) => $"TokenByIndex:{index}";
+
+    private ulong GetTokenByIndex(ulong index) => this.PersistentState.GetUInt64(GetTokenByIndexKey(index));
+
+    private void SetTokenByIndex(ulong index, ulong token) => this.PersistentState.SetUInt64(GetTokenByIndexKey(index), token);
+
+    private void ClearTokenByIndex(ulong index) => this.PersistentState.Clear(GetTokenByIndexKey(index));
+
+    private string GetIndexByTokenKey(ulong token) => $"IndexByToken:{token}";
+
+    private ulong GetIndexByToken(ulong token) => this.PersistentState.GetUInt64(GetIndexByTokenKey(token));
+
+    private void SetIndexByToken(ulong token, ulong index) => this.PersistentState.SetUInt64(GetIndexByTokenKey(token), index);
+
+    private void ClearIndexByToken(ulong token) => this.PersistentState.Clear(GetIndexByTokenKey(token));
+
+    private string GetTokenOfOwnerByIndexKey(Address address, ulong index) => $"TokenOfOwnerByIndex:{address}:{index}";
+
+    private ulong GetTokenOfOwnerByIndex(Address address, ulong index) => this.PersistentState.GetUInt64(GetTokenOfOwnerByIndexKey(address, index));
+
+    private void SetTokenOfOwnerByIndex(Address owner, ulong index, ulong tokenId) => this.PersistentState.SetUInt64(GetTokenOfOwnerByIndexKey(owner, index), tokenId);
+
+    private void ClearTokenOfOwnerByIndex(Address owner, ulong index) => this.PersistentState.Clear(GetTokenOfOwnerByIndexKey(owner, index));
+
+    private string IndexOfOwnerByTokenKey(Address owner, ulong tokenId) => $"IndexOfOwnerByToken:{owner}:{tokenId}";
+    private ulong GetIndexOfOwnerByToken(Address owner, ulong tokenId) => this.PersistentState.GetUInt64(IndexOfOwnerByTokenKey(owner, tokenId));
+    private void SetIndexOfOwnerByToken(Address owner, ulong tokenId, ulong index) => this.PersistentState.SetUInt64(IndexOfOwnerByTokenKey(owner, tokenId), index);
+    private void ClearIndexOfOwnerByToken(Address owner, ulong tokenId) => this.PersistentState.Clear(IndexOfOwnerByTokenKey(owner, tokenId));
```

Comment:
related to this comment: https://github.com/stratisproject/CirrusSmartContracts/pull/28#discussion_r565821159

I think all of the `Get` methods in this selection should be public.

Reply:
There are public methods already provide needed functionality. These are just getter type of methods properties and public method contains an extra logic over them.